PR: https://github.com/odoo/odoo/pull/47344

From: be7d1d7510bc94f949a4fc55a6038a83fb026147
From: jerome hanke (jhk)
Date: 2020-03-11 09:12:59

Breaking data model changes score: 3, change matches:
-    analytic_account_id = fields.Many2one(compute="_compute_analytic_account", store=True, readonly=False)
-    analytic_tag_ids = fields.Many2many(compute="_compute_analytic_account", store=True, readonly=False)

Total Changes: 8

[FIX] account_analytic_default: computed fields cannot be readonly=False

Steps to reproduce:
* install acounting, make sure you have account_analytic_default installed
* go to settings and activate Analytic Accounting
* go to accounting > configuration > analytic accounts > create an account
* go to accounting > customers > invoices > create > make an invoice line with
an analytical account > save > edit > add a new line to the invoice and select
any product

Previous behavior:
analytic accounts previously saved are deleted by an onchange

Current behavior:
analytic accounts remains set after any update of the invoice lines

opw-2209169

closes odoo/odoo#47344

Signed-off-by: Nicolas Martinelli (nim) <nim@odoo.com>

================================= pseudo patch: =================================

--- a/addons/account_analytic_default/models/account_analytic_default.py
+++ b/addons/account_analytic_default/models/account_analytic_default.py
@@ -68,12 +68,8 @@ class AccountAnalyticDefault(models.Model):
 class AccountMoveLine(models.Model):
     _inherit = 'account.move.line'
 
-    # Overload of fields defined in account
-    analytic_account_id = fields.Many2one(compute="_compute_analytic_account", store=True, readonly=False)
-    analytic_tag_ids = fields.Many2many(compute="_compute_analytic_account", store=True, readonly=False)
-
-    @api.depends('product_id', 'account_id', 'partner_id', 'date_maturity')
-    def _compute_analytic_account(self):
+    @api.onchange('product_id', 'account_id', 'partner_id', 'date_maturity')
+    def _onchange_analytic_account(self):
         for record in self:
             record.analytic_account_id = record.analytic_account_id or False
             record.analytic_tag_ids = record.analytic_tag_ids or False
